gateway/Cargo.lock

```diff
@@ -0,0 +1,836 @@
+# This file is automatically @generated by Cargo.
```

Comment:
Can this be created automatically? If yes, can we autogenerate is rather than commit it

Reply:
@wagmarcel it's recommended to keep `Cargo.lock` file for binaries and ignore it for libs

https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html